samplers/aws/xray/clock.go

```diff
@@ -0,0 +1,32 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package xray
+
+import (
+	"time"
+)
+
+// Clock provides an interface to implement method for getting current time.
+type Clock interface {
+	Now() time.Time
+}
+
+// DefaultClock is an implementation of Clock interface.
+type DefaultClock struct{}
+
+// Now returns current time.
+func (t *DefaultClock) Now() time.Time {
+	return time.Now()
+}
```

Comment:
These look to only exist to allow testing overrides. They do not look needed by the user. Please unexport them or move to an internal package.